receiver/zipkinreceiver/config.go

```diff
@@ -4,22 +4,71 @@
 package zipkinreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+var disallowHTTPDefaultProtocol = featuregate.GlobalRegistry().MustRegister(
+	"zipkinreceiver.httpDefaultProtocol.disallow",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterDescription("When enabled, usage of the default http configuration is disallowed"),
+	featuregate.WithRegisterFromVersion("v0.114.0"),
 )
 
+const deprecationConfigMsg = "the inline setting of http server parameters has been deprecated, please use .protocols.http parameter instead."
+
 // Config defines configuration for Zipkin receiver.
 type Config struct {
 	// Configures the receiver server protocol.
+	//
+	// Deprecated: Parameter exists for historical compatibility
+	// and should not be used. To set the server configurations,
+	// use the Protocols parameter instead.
 	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 	// If enabled the zipkin receiver will attempt to parse string tags/binary annotations into int/bool/float.
 	// Disabled by default
 	ParseStringTags bool `mapstructure:"parse_string_tags"`
+	// Protocols define the supported protocols for the receiver
+	Protocols ProtocolTypes `mapstructure:"protocols"`
+}
+
+type ProtocolTypes struct {
+	HTTP confighttp.ServerConfig `mapstructure:"http"`
 }
 
 var _ component.Config = (*Config)(nil)
 
 // Validate checks the receiver configuration is valid
 func (cfg *Config) Validate() error {
+	if isServerConfigDefined(cfg.ServerConfig) {
+		if disallowHTTPDefaultProtocol.IsEnabled() {
+			return fmt.Errorf(deprecationConfigMsg)
+		}
+		if isServerConfigDefined(cfg.Protocols.HTTP) {
+			return fmt.Errorf("cannot use .protocols.http together with default server config setup")
```

Comment:
I'm afraid that with the current implementation of this change, the collector will fail to start also when user does not specify the `protocols::http` configuration. This is because the `protocols::http` configuration is part of the default config. As a result, collector will fail to start with this config:

```yaml
receivers:
  zipkin:
    endpoint: localhost:1234
```

To fix this, I propose to follow the approach that the OTLP receiver does [here](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.114.0/receiver/otlpreceiver/config.go#L72-L73). It implements the `Unmarshal` method on the config where it removes the part of the default config that was not set by the user (the `conf` structure only contains config options set by the user).